File: MNA_Methods/MATLAB/2-General MNA/original_system.m
%---------------------------------------------------------------------------------------------------------------------------------------%

% original_system.m
% 
% Author: Rasul Choupanzadeh
% Date: 05/12/2022
% 
% 
% This code is based on the concepts from [1-2].
% 
% [1] Yangfeng Su, Jian Wang, Xuan Zeng, Zhaojun Bai, C. Chiang, and D. Zhou.
%     SAPOR: second-order arnoldi method for passive order reduction of RCS circuits.
%     In IEEE/ACM International Conference on Computer Aided Design (ICCAD), pages 74–79, 2004.
% 
% [2] Bang Liu, Xuan Zeng, Yangfeng Su, Jun Tao, Zhaojun Bai, C. Chiang, and Dian Zhou.
%     Block SAPOR: block second-order arnoldi method for passive order reduction of multi-input multi-output 
%     RCS interconnect circuits. In Proceedings of Asia and South Pacific Design Automation Conference (ASP-DAC), pages 244–249, 2005.

%---------------------------------------------------------------------------------------------------------------------------------------%



V_orig=zeros(q,length(f));
nt=1;
for freq=fmin:fstep:fmax
    s=1i*2*pi*freq*Coef;
    V_orig(:,nt)=transpose(Lv)*((s.*Cm+Gm+Gamma./s)\Bm)*Jm;
    nt=nt+1;
end 

clear nt freq s 

File: MNA_Methods/Python/2-General MNA/original_system.py

''' original_system.py

Author: Rasul Choupanzadeh
Date: 05/12/2022


This code is based on the concepts from [1-2].

[1] Yangfeng Su, Jian Wang, Xuan Zeng, Zhaojun Bai, C. Chiang, and D. Zhou.
    SAPOR: second-order arnoldi method for passive order reduction of RCS circuits.
    In IEEE/ACM International Conference on Computer Aided Design (ICCAD), pages 74–79, 2004.

[2] Bang Liu, Xuan Zeng, Yangfeng Su, Jun Tao, Zhaojun Bai, C. Chiang, and Dian Zhou.
    Block SAPOR: block second-order arnoldi method for passive order reduction of multi-input multi-output 
    RCS interconnect circuits. In Proceedings of Asia and South Pacific Design Automation Conference (ASP-DAC), pages 244–249, 2005.

'''



# Calculate the original frequency response    
start = time.time()
V_orig = np.zeros(shape=(q,len(freq)), dtype = 'complex')
nt = 0
for f in freq:        
    s = 1j*2*np.pi*f*Coef
    V = np.transpose(Lv) @ (np.linalg.inv(s*Cm + Gm + Gamma/s) @ Bm) @ Jm
    V_orig[:,nt]=V[:,0]
    nt = nt+1

end = time.time()


